Content.Shared/Xenoarchaeology/Artifact/SharedXenoArtifactSystem.Unlock.cs

```diff
@@ -72,7 +73,21 @@ public void FinishUnlockingState(Entity<XenoArtifactUnlockingComponent, XenoArti
         XenoArtifactUnlockingComponent unlockingComponent = ent;
 
         SoundSpecifier? soundEffect;
-        if (TryGetNodeFromUnlockState(ent, out var node))
+
+        // Frontier: Disable activations on protected grids
+        var xform = Transform(ent);
+        var gridProtected = false;
+        if (xform.GridUid != null)
+        {
+            if (TryComp<ProtectedGridComponent>(xform.GridUid.Value, out var prot) && prot.PreventArtifactTriggers)
+            {
+                gridProtected = true;
+            }
+        }
+        // End Frontier: Disable activations on protected grids
+
+
+        if (TryGetNodeFromUnlockState(ent, out var node) && !gridProtected) // Frontier, add grid protection check
```

Comment:
```suggestion
        if (!gridProtected && TryGetNodeFromUnlockState(ent, out var node)) // Frontier, add grid protection check
```
You're already assigned the value to check, might as well check it first.